File: Python/2824.py
# 2824. Count Pairs Whose Sum is Less than Target
# https://leetcode.com/problems/count-pairs-whose-sum-is-less-than-target/

class Solution:
  def countPairs(self, nums: List[int], target: int) -> int:
    out = 0

    for i in range(0, len(nums) - 1):
      for j in range(i + 1, len(nums)):
        if (nums[i] + nums[j] < target):
          out += 1

    return out

File: Rust/2824.rs
/**
 * 2824. Count Pairs Whose Sum is Less than Target
 * https://leetcode.com/problems/count-pairs-whose-sum-is-less-than-target/
 */

impl Solution {
  pub fn count_pairs(nums: Vec<i32>, target: i32) -> i32 {
    let mut out: i32 = 0;

    for i in 0..nums.len() - 1 {
      for j in i + 1..nums.len() {
        if nums[i] + nums[j] < target {
          out += 1;
        }
      }
    }

    return out;
  }
}
